source/ice_api.c

```diff
@@ -0,0 +1,1676 @@
+        ( pSrcAddr == NULL ) ||
+        ( pTransactionIdBuffer == NULL ) )
+    {
+        retStatus = ICE_RESULT_BAD_PARAM;
+    }
+
+    if( retStatus == ICE_RESULT_OK )
+    {
+        retStatus = Ice_InitializeStunPacket( pIceAgent,
+                                              &stunCxt,
+                                              pTransactionIdBuffer,
+                                              &pIceAgent->stunMessageBuffer[ 0 ],
+                                              &stunHeader,
+                                              0,
+                                              0 );
+    }
+
+    if( retStatus == ICE_RESULT_OK )
+    {
+        memcpy( &stunMappedAddress,
+                &pSrcAddr->ipAddress,
+                sizeof( StunAttributeAddress_t ) );
+        stunRetStatus = StunSerializer_AddAttributeXorMappedAddress( &stunCxt,
+                                                                     &stunMappedAddress );
+    }
+
+    if( stunRetStatus == STUN_RESULT_OK )
+    {
+        if( pIceAgent->isControlling == 0 )
+        {
+            stunRetStatus = StunSerializer_AddAttributeIceControlled( &stunCxt,
+                                                                      pIceAgent->tieBreaker );
+        }
+        else
+        {
+            stunRetStatus = StunSerializer_AddAttributeIceControlling( &stunCxt,
+                                                                       pIceAgent->tieBreaker );
+        }
+    }
+
+    if( stunRetStatus == STUN_RESULT_OK )
+    {
+        retStatus = Ice_PackageStunPacket( pIceAgent,
+                                           &stunCxt,
+                                           ( uint8_t * ) pIceAgent->localPassword,
+                                           ( uint32_t ) strlen( pIceAgent->localPassword ) * sizeof( char ),
+                                           pSendStunMessageBufferLength );
+    }
+
+    if( retStatus == ICE_RESULT_OK )
+    {
+        *ppSendStunMessageBuffer = &pIceAgent->stunMessageBuffer[ 0 ];
+    }
+
+    return retStatus;
+}
+/*------------------------------------------------------------------------------------------------------------------*/
+
+/* Ice_DeserializeStunPacket - This API deserializes a received STUN packet . */
+
+IceResult_t Ice_DeserializeStunPacket( StunContext_t * pStunCxt,
+                                       StunHeader_t * pStunHeader,
+                                       StunAttribute_t * pStunAttribute,
+                                       IceStunDeserializedPacketInfo_t * pDeserializedPacketInfo )
+{
+    IceResult_t retStatus = ICE_RESULT_OK;
+    StunResult_t stunRetStatus = STUN_RESULT_OK;
+    uint8_t * pErrorPhase;
+    uint16_t errorPhaseLength;
+
+    if( ( pStunCxt == NULL ) ||
+        ( pStunHeader == NULL ) ||
+        ( pStunAttribute == NULL ) ||
+        ( pDeserializedPacketInfo == NULL ) )
+    {
+        retStatus = ICE_RESULT_BAD_PARAM;
+    }
+
+    /* Initialization */
```

Comment:
Should be inside if( retStatus = ICE_RESULT_OK)

Reply:
Added inside an `else` block